src/providers/incomplete-tests/incomplete-tests.spec.ts

```diff
@@ -0,0 +1,187 @@
+      },
+    },
+  };
+
+  const testStatuses: { [slotId: string]: TestStatus } = {
+    1001: TestStatus.Booked,
+    1003: TestStatus.Completed,
+    1005: TestStatus.Decided,
+    1007: TestStatus.Started,
+    1008: TestStatus.Submitted,
+  };
+
+  const slots: SlotItem[] = [
+    {
+      slotData: {
+        booking: {
+          application: {
+            testCategory: 'B',
+          },
+        },
+        slotDetail: {
+          slotId: 1001,
+          start: '2019-06-26T11:11:00+01:00',
+        },
+        vehicleSlotType: 'B57mins',
+      },
+      hasSlotChanged: false,
+    },
+    {
+      slotData: {
+        booking: {
+          application: {
+            testCategory: 'B',
+          },
+        },
+        slotDetail: {
+          slotId: 1003,
+          start: '2019-06-25T11:11:00+01:00',
+        },
+        vehicleSlotType: 'B57mins',
+      },
+      hasSlotChanged: false,
+    },
+    {
+      slotData: {
+        booking: {
+          application: {
+            testCategory: 'B',
+          },
+        },
+        slotDetail: {
+          slotId: 1005,
+          start: '2019-06-25T11:11:00+01:00',
+        },
+        vehicleSlotType: 'B57mins',
+      },
+      hasSlotChanged: false,
+    },
+    {
+      slotData: {
+        booking: {
+          application: {
+            testCategory: 'B',
+          },
+        },
+        slotDetail: {
+          slotId: 1007,
+          start: '2019-06-27T11:11:00+01:00',
+        },
+        vehicleSlotType: 'B57mins',
+      },
+      hasSlotChanged: false,
+    },
+    {
+      slotData: {
+        booking: {
+          application: {
+            testCategory: 'B',
+          },
+        },
+        slotDetail: {
+          slotId: 1008,
+          start: '2019-06-26T11:11:00+01:00',
+        },
+        vehicleSlotType: 'B57mins',
+      },
+      hasSlotChanged: false,
+    }];
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        IncompleteTestsProvider,
+        { provide: SlotSelectorProvider, useClass: SlotSelectorProvider },
+        { provide: AppConfigProvider, useClass: AppConfigProviderMock },
+      ],
+    });
+    incompleteTestsProvider = TestBed.get(IncompleteTestsProvider);
+  });
+
+  describe('canStartTest', () => {
+    it('should return true when a cat B test is used', async () => {
+      const result = incompleteTestsProvider.canStartTest(catBTest);
+      expect(result).toBe(true);
+    });
+    it('should return false when a cat B+E test is used', async () => {
+      const result = incompleteTestsProvider.canStartTest(catBETest);
+      expect(result).toBe(false);
+    });
+  });
+
+  describe('countIncompleteTests', () => {
+    it('should return 2 tests in the past that arent submitted when passed the mock data', () => {
+      const result = incompleteTestsProvider.countIncompleteTests(testStatuses, slots);
+      expect(result).toEqual(2);
+    });
+  });
+
+  describe('isDateInPast', () => {
```

Comment:
Do you intend `isDateInPast` to be part of the public API of the `IncompleteTestsProvider`? Or is it just a helper method? If the latter, I'd say put `private` on it and delete these tests, the tests for `countIncompleteTests` should cover it.